### PR TITLE
Refactor DatasetOrTimeSchedule timetable docs

### PR DESCRIPTION
**Pull request by vatsrahul1001**
As we have changed timetable name from `DatasetTimetable`  to `DatasetOrTimeSchedule` , looks like docs were not updated fixing that in this PR

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow/authoring-and-scheduling/timetable.rst [hunk 1]
```diff
@@ -179,34 +179,15 @@ first) event for the data interval, otherwise manual runs will run with a ``data
 
 .. _dataset-timetable-section:
 
-DatasetTimetable
-^^^^^^^^^^^^^^^^
+Integrate conditional dataset with Time-Based Scheduling
```
Comment (dirrao) [inline docs/apache-airflow/authoring-and-scheduling/timetable.rst]: Can you enhance it better? 
may be "Dataset event based scheduling with time based scheduling"

Comment (sunank200) [inline docs/apache-airflow/authoring-and-scheduling/timetable.rst]:
```suggestion
Dataset event based scheduling with time based scheduling
```

**Review by sunank200 — APPROVED** [review]

**Review by dirrao — APPROVED** [review]
LGTM

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
docs/static check failures

**Comment by vatsrahul1001** [comment]
> docs/static check failures

resolved

**Review by potiuk — APPROVED** [review]

**Pull request merged**